dts/arm/st/stm32f0.dtsi

```diff
@@ -151,14 +172,17 @@
 		};
 
 		spi1: spi@40013000 {
+			spi-controller;
```

Comment:
Unfortunately use of spi-controller is  not well documented, neither in Linux Documentation or Device tree specification.
At least I think it is not safe to base the generation solution on the hypothesis each node is a thing-controller

Reply:
Ok. Can we go for node_type and have the controller type in a generic node type that has to be included (eg. spi.yaml, i2c.yaml, ...)?

Reply:
Sounds like a possible alternative. I introduced node_type due to lack of generic name available,
and the feeling this would be required eventually. But if this is to be used in codegen snippets, I think ideal would be to refer to device tree element like node name (which is basically a zephyr device) or compatible (which is the driver that will be used to control the device)